src/main/java/edu/kis/powp/jobs2d/transformations/LinesTransformationExecutor.java

```diff
@@ -0,0 +1,56 @@
+package edu.kis.powp.jobs2d.transformations;
+
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.powp.jobs2d.features.LinesRecorder;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+import edu.kis.powp.jobs2d.features.WorkspaceTransformationRecorder;
+
+import java.awt.Point;
+import java.util.List;
+
+public class LinesTransformationExecutor {
+    private final LinesRecorder linesRecorder;
+
+    public LinesTransformationExecutor() {
+        this.linesRecorder = LinesRecorder.getLinesRecorder();
+    }
+
+    public void execute(Transformation transformation) {
+        DrawerFeature.getDrawerController().clearPanel();
+
+        List<ILine> lines = linesRecorder.getLines();
+        List<ILine> untransformedLines = linesRecorder.getUntransformedLines();
+
+        for (int i = 0; i < lines.size(); i++) {
+            ILine line = lines.get(i);
+            ILine untransformedLine = untransformedLines.get(i);
+
+            if (transformation instanceof ScaleTransformation) {
```

Comment:
Not Object-Oriented way. Avoiding instanceof is probably possible (maybe Strategy or Visitor?)